logging/src/commonMain/kotlin/DispatchLogger.kt

```diff
@@ -1,44 +1,51 @@
 package com.juul.tuulbox.logging
 
+import co.touchlab.stately.isolate.IsolateState
+
 /** Implementation of [Logger] which dispatches calls to consumer [Logger]s. */
 public class DispatchLogger : Logger {
-    private val consumers = mutableSetOf<Logger>()
+    private val consumers = IsolateState { mutableSetOf<Logger>() }
 
     /** `false` if no consumers have been installed, `true` if at least one consumer has been installed. */
     internal val hasConsumers: Boolean
-        get() = consumers.isNotEmpty()
+        get() = consumers.access { it.isNotEmpty() }
 
     /** Add a consumer to receive future dispatch calls. */
     public fun install(consumer: Logger) {
-        consumers.add(consumer)
+        consumers.access { it.add(consumer) }
     }
 
     /** Uninstall all installed consumers. */
     public fun clear() {
-        consumers.clear()
+        consumers.access { it.clear() }
+    }
+
+    /** Dispose of this dispatch logger when it is no longer needed. */
+    public fun dispose() {
+        consumers.dispose()
     }
 
     override fun verbose(tag: String, message: String, throwable: Throwable?) {
-        consumers.forEach { it.verbose(tag, message, throwable) }
+        consumers.access { it.forEach { it.verbose(tag, message, throwable) } }
```

Comment:
We may be able to avoid the context switching (that I believe occurs with Stately) if we store a copy of `consumers` in another property?

In other words, we have a collection (as you have it now) that we populate, each time it is modified (via Stately) we also save a copy to an `atomic` property. The copy will be frozen (which is fine) but then we can read (and iterate) over that read-only (frozen) copy on any thread (and not have to do the context switching for every `debug`, `info`, etc call).

Reply:
Ended up going with a full list copy every time we add a new consumer to keep the implementation more simple. In theory this drops to like `O(n^2)` efficiency on adding new loggers, but `n` should always be small enough here I'm not worried about it.